### PR TITLE
Changes missed out by PR #492

### LINKED ISSUE #488: MCM to mark terminating nodes
**What would you like to be added**:

When MCM replaces nodes (unhealthy node termination, node rolling updates, ...), nodes that are being terminated should be marked in some way (e.g. annotation) to reflect the reason of their termination. 

**Why is this needed**:
End-users can setup monitoring for the progress of node rolling updates. If PDBs are used, end-users can automate removal of PDBs from pods that are running on nodes which are being terminated due to health issues.

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:
Add termination condition to deleting nodes on OOT. This fixes missed out changes for node conditions update on OOT.

**Which issue(s) this PR fixes**:
Fixes #488 

**Special notes for your reviewer**:
I have split the changes into two commits here - https://github.com/gardener/machine-controller-manager/pull/505/commits.
1. Refactors existing code.
2. Makes these changes on OOT code path.

**Release note**:

```improvement operator
NONE
```

/priority critical
/kind regression

### CODE REVIEW

File: pkg/util/provider/machinecontroller/machine_util.go [hunk 1]
```diff
@@ -889,50 +890,67 @@ func (c *controller) drainNode(deleteMachineRequest *driver.DeleteMachineRequest
...
-		err = drainOptions.RunDrain()
-		if err == nil {
-			// Drain successful
-			klog.V(2).Infof("Drain successful for machine %q. \nBuf:%v \nErrBuf:%v", machine.Name, buf, errBuf)
+		// update node with the machine's state prior to termination
+		if err = c.UpdateNodeTerminationCondition(machine); err != nil {
+			if forceDeleteMachine {
+				klog.Warningf("Failed to update node conditions: %v. However, since it's a force deletion shall continue deletion of VM.", err)
+			} else {
+				klog.Errorf("Drain failed due to failure in update of node conditions: %v", err)
```
Comment (hardikdr) [inline pkg/util/provider/machinecontroller/machine_util.go:898]: Should we return error on failure here ?
- Original PR https://github.com/gardener/machine-controller-manager/pull/492/files#diff-d8287fe74b5273163c9f6b6c635ad912R609

**Review by hardikdr — COMMENTED** [review]
Just a minor comment, looks overall.

File: pkg/util/provider/machinecontroller/machine_util.go (hunk 1 above)
Comment (prashanth26) [reply pkg/util/provider/machinecontroller/machine_util.go:898]: I have avoided returning error here to make sure that the machine object status is updated on failure here - https://github.com/gardener/machine-controller-manager/blob/e073a81ab725d699a89044876ce8c1bfbb2eff33/pkg/util/provider/machinecontroller/machine_util.go#L900-L904.

At the end of the function the error is still returned here - https://github.com/gardener/machine-controller-manager/blob/e073a81ab725d699a89044876ce8c1bfbb2eff33/pkg/util/provider/machinecontroller/machine_util.go#L975

**Review by hardikdr — APPROVED** [review]
Thanks for the nice PR, looks good.
/lgtm

**Pull request merged**